### PR TITLE
feat(search): add snapshoted latest version and date of a package

### PR DESCRIPTION
**Pull request by 94noni**
Hello,

Small PR related to issues mentioned/related.

> [!NOTE]
> PR opened as is to gather feedbacks/have raw material for discussions (not finished at all)
> I am open to discuss the easiest/fastest/simplest way for introducing such data that will help a think a lot the search results

---

## Idea of display

Capture d’écran 2024-01-08 à 18 47 42

---

refs:

- https://github.com/composer/packagist/issues/1320 (old one of mine)
- https://github.com/composer/packagist/issues/1290
- https://github.com/composer/packagist/issues/1368

### CODE REVIEW

File: src/Entity/Package.php [hunk 1]
```diff
@@ -539,6 +542,14 @@ public function getBrowsableRepository(): string
     public function addVersion(Version $version): void
     {
         $this->versions[] = $version;
+
+        $newVersion = strtolower($version->getNormalizedVersion());
+
+        if (null === $this->snapshotedLatestVersion
+            ||
+            version_compare($newVersion, $this->snapshotedLatestVersion, '>=')) {
```
Comment (stof) [inline src/Entity/Package.php:550]: `version_compare` looks wrong to me here given that you don't only include the version when writing in the property.

File: src/Entity/Package.php [hunk 2]
```diff
@@ -539,6 +542,14 @@ public function getBrowsableRepository(): string
...
+            $this->snapshotedLatestVersion = $newVersion . '&' . (new \DateTime('now'))->format('Y-m-d');
```
Comment (stof) [inline src/Entity/Package.php:551]: the date of adding the data in the entity is not the date of that version (it depends when the sync happens)

File: js/search.js [hunk 3]
```diff
@@ -174,6 +174,8 @@ search.addWidget(
         <div class="col-sm-3 col-lg-2">
             {{#meta}}
                 <p class="metadata">
+                    <span class="metadata-block"><i class="glyphicon glyphicon-tags"></i> {{ meta.latest_version }}</span>
+                    <span class="metadata-block"><i class="glyphicon glyphicon-calendar"></i> {{ meta.latest_date }}</span>
```
Comment (stof) [inline js/search.js:178]: what if those are not available ?

File: src/Entity/Package.php (hunk 2 above)
Comment (94noni) [reply src/Entity/Package.php:551]: indeed! we may leverage `Version::releasedAt` here?

File: js/search.js (hunk 3 above)
Comment (94noni) [reply js/search.js:178]: i have not install/run the project locally for now, dont want to invest much time of PR is not accepted
so for now, it was just to make a decent screenshot with fake data ^^

File: src/Entity/Package.php (hunk 1 above)
Comment (94noni) [reply src/Entity/Package.php:550]: indeed it was my original idea, 2 fields, but for snapshots data I went for one
i handle here the explode

**Pull request closed without merging**